pilot/pkg/model/context.go

```diff
@@ -66,6 +66,8 @@ type Environment struct {
 	// START OF THE PUSH, THE GLOBAL ONE MAY CHANGE AND REFLECT A DIFFERENT
 	// CONFIG AND PUSH
 	PushContext *PushContext
+
+	StatusReporter DistributionEventHandler
```

Comment:
Is it really part of 'model' ? The purpose was to have interface and common structs, not all features.

Reply:
I was not able to find any documentation on the original intent of the model package.  If there is a more idiomatic way to allow ads.go to have access to the EventHandler, please let me know.

Reply:
Sure, keep it in DiscoveryService or Server.

Reply:
Refactored to DiscoveryService.